app/common/auth/auth.service.ts

```diff
@@ -0,0 +1,79 @@
+import 'rxjs/Rx';
+
+import {Injectable} from '@angular/core';
+import {CustomHttp} from "../services/CustomHttp";
+import {AuthData} from "./auth.data";
+import {Router} from "@angular/router";
+import {Observable} from "rxjs/Rx";
+
+@Injectable()
+export class AuthService {
```

Comment:
One more thing I didn't mention: do not produce much code for work with users, their authority and roles! Such service must appear a long ago !